### PR TITLE
Replace OkHttp with java Http

### LINKED ISSUE #4: Use java Http requests
Use java Http request instead of OkHttp, reasons:-
- Java have the feature, no need for external lib

OkHttp was used before coz, this project was written in java 8, and java http lib came in java 11. But now java version is updated.

### PR DESCRIPTION
**Pull request by java-coding-prodigy**
Fixes #4 

One issue I am facing is the entire `CookieJar` system throughout the IG API, java http does not have a straightforward replacement to that class, and we might have to make a few utility classes to create support for the OkHttp cookie handling system

### CODE REVIEW

File: src/main/java/com/github/instagram4j/instagram4j/utils/IGUtils.java [hunk 1]
```diff
@@ -37,12 +39,14 @@ public class IGUtils {
...
     private static final ObjectWriter WRITER = MAPPER.writer();
+
     static {
         MAPPER.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
         MAPPER.setSerializationInclusion(Include.NON_NULL);
     }
 
-    private IGUtils() {}
+    private IGUtils() {
+    }
```
Comment (Taz03) [inline src/main/java/com/github/instagram4j/instagram4j/utils/IGUtils.java:49]: no don't change the formatting for now, we'll discuss on that later

File: src/main/java/com/github/instagram4j/instagram4j/utils/IGUtils.java [hunk 2]
```diff
@@ -52,7 +56,7 @@ private IGUtils() {}
     /**
      * Digest a string using the given codec and input
      *
-     * @param codec Codec to use
+     * @param codec  Codec to use
```
Comment (Taz03) [inline src/main/java/com/github/instagram4j/instagram4j/utils/IGUtils.java:59]: .

File: src/main/java/com/github/instagram4j/instagram4j/utils/IGUtils.java [hunk 3]
```diff
@@ -79,7 +83,7 @@ public static String md5hex(String source) {
     /**
      * Convert the byte array to a hexadecimal presentation (String)
      *
-     * @param bytes byte array
+     * @param bytes        byte array
```
Comment (Taz03) [inline src/main/java/com/github/instagram4j/instagram4j/utils/IGUtils.java:86]: .

File: src/main/java/com/github/instagram4j/instagram4j/utils/IGUtils.java [hunk 4]
```diff
@@ -135,7 +139,7 @@ public static String generateSignature(String payload) {
      * Converts an Instagram ID to their shortcode system.
      *
      * @param code The ID to convert. Must be provided as a string if it's larger than the size of
-     *        an integer, which MOST Instagram IDs are!
+     *             an integer, which MOST Instagram IDs are!
```
Comment (Taz03) [inline src/main/java/com/github/instagram4j/instagram4j/utils/IGUtils.java:142]: .

**Review by Taz03 — CHANGES_REQUESTED** [review]

**Comment by Taz03** [comment]
@java-coding-prodigy i'll suggest u to wait a week or two to get the tests fixed first, so it'll be easy to see if everything is ok

**Comment by Taz03** [comment]
also the cookie thing is correct, its hard to do with java's http reqs, so first come up with a solution for that and then start implementing, or remove it completely if possible

**Comment by java-coding-prodigy** [comment]
> also the cookie thing is correct, its hard to do with java's http reqs, so first come up with a solution for that and then start implementing, or remove it completely if possible

So should I make my own utility class?
Or maybe find some other modern API eith good cookie handling?

**Comment by Taz03** [comment]
@java-coding-prodigy will [CookieManager](https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/net/CookieManager.html) help? also i'll recommend to not work in this for a while. I might delete the project in favor of a complete rewrite. I'll ping u when things are decided

**Comment by java-coding-prodigy** [comment]
I am using CookieManager only for the cookie handling, but I have to do some ugly casting from CookieHandler to manager as HttpClient returns handler but only manager has the `cookieStore()` method which is needed, shouldn't be a problem as all client builders are only using manager but still


Also for the rewrite even I think it might be necessary, migrating from okhttp3 to java http itself seems like it needs a rewrite, so keep me updated on the status.

**Comment by Taz03** [comment]
@java-coding-prodigy rewrite it is, im trying to get the basic structure down then will create a new repo

**Comment by java-coding-prodigy** [comment]
Cool, I'd like to work on some of the networking though, if possible

**Comment by Taz03** [comment]
@java-coding-prodigy r u on discord?

**Comment by java-coding-prodigy** [comment]
> yes, I am `NotNull`

**Comment by Taz03** [comment]
bruh didn't knew

**Pull request closed without merging**